conans/client/installer.py

```diff
@@ -454,3 +456,15 @@ def _call_package_info(self, conanfile, package_folder, is_editable):
                 self._hook_manager.execute("post_package_info", conanfile=conanfile)
 
         conanfile.cpp_info.check_component_requires(conanfile)
+
+    def _call_install_method(self, conanfile, install_folder):
+        if hasattr(conanfile, "install"):
+            conanfile.folders.set_install_folder(install_folder)
+            if not os.path.exists(install_folder):
```

Comment:
This if is important, it means that the install() method will only be ran once per package id. This 1:1 correspondence is kept by ensruing only `self.info` is available from the `install()` method